libs/qCC_io/ShpFilter.cpp

```diff
@@ -893,10 +1044,18 @@ static CC_FILE_ERROR LoadCloud(QDataStream &shpStream,
 				ScalarType s = isESRINoData(m) ? NAN_VALUE : static_cast<ScalarType>(m);
 				sf->addElement(s);
 			}
-			sf->computeMinAndMax();
-			int sfIdx = cloud->addScalarField(sf);
-			cloud->setCurrentDisplayedScalarField(sfIdx);
-			cloud->showSF(true);
+			bool allNans = std::all_of(sf->begin(), sf->end(), [](ScalarType s) { return std::isnan(s); });
```

Comment:
Would this be useful to add to ccScalarField?

(Not suggesting it for this PR - just asking.)

Reply:
Maybe, the check has to be made because CC doesn’t like scalar fields full of NaNs, so we  need to make sure the scalar field contains a valid value before attaching it to an object

Reply:
Or we could add a more generic version that takes a scalar value as input? (but I agree, not in this PR)